Source interface UX improvements

redshiftzero opened the pull request:
These are a couple of small changes to the source interface suggested by @ninavizz in her UX feedback and mockup of the source interface shown [here](https://github.com/freedomofpress/securedrop/issues/1490#issuecomment-270820226). 

Adds a "Cancel" button (from UX feedback: "Giving users a cancel option reduces anxiety"):

![explicit-cancel-button](https://cloud.githubusercontent.com/assets/7832803/22358494/6bc6c876-e3f4-11e6-8c03-2453a5c17f8b.png)

Displays which journalist is speaking to the source on the source interface (when you're on the source side, it's a little confusing to know who you are talking to in the news org - is it the point person/admin, journalist Bob, journalist Alice??):

![which-journo](https://cloud.githubusercontent.com/assets/7832803/22358514/9e6db7bc-e3f4-11e6-992b-8a58e0cc1e44.png)

Also after #1523 is merged we could indeed display the full name and blurb as displayed in her mockup - happy to make that change when the time comes

ninavizz commented:
Could the "Cancel" button be next to the Submit button—not below it—and white background with dark text? All the submitted mockups reflect existing interaction paradigms users expect. Stacking the buttons is very odd and will be confusing. 
![screenshot_cancel-butt](https://cloud.githubusercontent.com/assets/8262612/22389287/a4dd8e56-e499-11e6-8ae5-c87fef321c69.png)

redshiftzero commented:
Thanks for this feedback @ninavizz! I'll modify this PR to make this change.

ninavizz commented:
No biggie—I'm creating a single new "issue" for all my 0.4 reccos, with each tweak numbered—so that you and @fowlslegs can more easily reference stuff in PRs. :)

redshiftzero commented:
Great - thanks for doing that @ninavizz!

fowlslegs commented:
I was going to make the same comment on the button location and color scheme, but @ninavizz beat me to it :smile_cat: It's such a simple PR, but it would still be nice to add a functional test that clicks it during a upload and makes sure it does it's job to prevent against future regressions.

redshiftzero commented:
I've modified this PR following the detailed guidance provided in #1536 and implemented a few more of the items from there:

* Item 9b: Change h1 text to "Submit Materials" and "Get Replies"
* Item 9d: Move GPG text to below the header, format it gray and left-aligned
* Item 9e: Change graphic from cloud to the data store icon made by @ninavizz
* Item 9f: Adds Cancel button next to the Submit button
* Item 9i: Change  “Log Out” to “Exit”
* Item 10a: Modify login text to "Enter codename"
* Item 10b: Add a "Cancel" button here too

Also added functional tests for the new "Cancel" buttons on `/login` and `/lookup`

Screenshots of the login and submit pages (note that the formatting of the button text as caps and the headlines is covered by @heartsucker's PR #1538):

![login](https://cloud.githubusercontent.com/assets/7832803/22524977/41a76bf6-e87a-11e6-8400-cd91cb7e14bd.png)

![submit](https://cloud.githubusercontent.com/assets/7832803/22525155/0016dd06-e87b-11e6-814a-087a102a77ed.png)

ninavizz commented:
Did you guys decide to punt on making all the task headers blue w/ grey-text as sub-headers? Sorry I made the instructions confusing by referencing what was really an H2 as an H1, and I later noticed I'd also spec'd the incorrect blue. `#7a84a4` is the correct wedgewood blue for the headers, and `#555` would work for the description text.

Other than those (dorky) nits, I'm super stoked to see how implementing the other recommendations are coming along—yeah!!! :)

redshiftzero commented:
No worries - and yep @heartsucker implemented the left aligning and blue coloring of headers in #1538 (so those changes won't appear in this PR)
